# Add macke check target to run tests based on Fedora's patch.
diff --git a/Makefile b/Makefile
index 3aed365d..f809f6ab 100644
--- a/Makefile
+++ b/Makefile
@@ -158,6 +158,11 @@ uninstall:
 	$(RMDIR) $(UNINSTALL_DIRS) || :
 	@echo "==== Successfully uninstalled LuaJIT $(VERSION) from $(PREFIX) ===="
 
+check: $(INSTALL_DEP)
+	@echo "==== Running tests for LuaJIT $(VERSION) ===="
+	$^ test/test.lua
+	@echo "==== All tests for LuaJIT $(VERSION) succeeded ===="
+
 ##############################################################################
 
 amalg:

# Update tests for latest LuaJIT Version based on Fedora's patch.
diff --git a/test/lib/contents.lua b/test/lib/contents.lua
index a1d8b9b..ca8391c 100644
--- a/test/lib/contents.lua
+++ b/test/lib/contents.lua
@@ -77,7 +77,8 @@ do --- 5.2 string +lua>=5.2
 end
 
 do --- pre-5.2 table +lua<5.2
-  check(table, "concat:foreach:foreachi:getn:insert:maxn:remove:sort", "pack:unpack:setn:new")
+  -- This changed in recent versions of LuaJIT
+  check(table, "concat:foreach:foreachi:getn:insert:maxn:move:remove:sort", "pack:unpack:setn:new")
 end
 
 do --- 5.2 table +lua>=5.2
@@ -118,9 +119,18 @@ end
 
 do --- pre-5.2 package +lua<5.2
   assert(package.loaders)
-  assert(not package.searchers)
+  -- This assert is known to fail with more recent versions of LuaJIT
+  -- assert(not package.searchers)
   assert(package.seeall)
 end
+	
+-- LuaJIT version check for lua will return true for +lua<5.2 since it
+-- does not fully implement 5.2.  Move the (not package.searchers) check
+-- to +compat5.2 instead of the version check since it is implemented by
+-- compat5.2.
+do --- 5.2 compat package +compat5.2
+  assert(package.searchers)
+end
 
 do --- 5.2 package +lua>=5.2
   assert(not package.loaders)

# Update test with undefined behavior.
diff --git a/test/lib/ffi/bit64.lua b/test/lib/ffi/bit64.lua
index d1b47be..08c9b05 100644
--- a/test/lib/ffi/bit64.lua
+++ b/test/lib/ffi/bit64.lua
@@ -18,7 +18,11 @@ end
 
 do --- smoke band
   assert(tostring(band(1ll, 1, 1ll, -1)) == "1LL")
-  assert(tostring(band(1ll, 1, 1ull, -1)) == "1ULL")
+  -- This test is known to fail with more recent versions of LuaJIT. See:
+  -- https://github.com/LuaJIT/LuaJIT/issues/837 for more information.
+  -- Patching to change -1 to -1ull to avoid undefined behavior mentioned in the
+  -- issue above.
+  assert(tostring(band(1ll, 1, 1ull, -1ull)) == "1ULL")
 end
 
 do --- smoke shl
